### PR TITLE
Fix EventSource for B3Propagator

### PR DESCRIPTION
**Pull request by cijothomas**
Own eventsource.
Also moved the public api files to unshipped, as the stable has not been released yet

### CODE REVIEW

File: src/OpenTelemetry.Extensions.Propagators/OpenTelemetryPropagatorsEventSource.cs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+using System.Diagnostics.Tracing;
+
+namespace OpenTelemetry.Internal
+{
+    /// <summary>
+    /// EventSource implementation for OpenTelemetry Propagators.
+    /// This is used for internal logging of this library.
+    /// </summary>
+    [EventSource(Name = "OpenTelemetry.Extensions.Propagators")]
+    internal class OpenTelemetryPropagatorsEventSource : EventSource
```
Comment (CodeBlanch) [inline src/OpenTelemetry.Extensions.Propagators/OpenTelemetryPropagatorsEventSource.cs:27]: nit: `internal sealed class`

File: src/OpenTelemetry.Extensions.Propagators/OpenTelemetryPropagatorsEventSource.cs [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    {
+        public static OpenTelemetryPropagatorsEventSource Log = new();
```
Comment (CodeBlanch) [inline src/OpenTelemetry.Extensions.Propagators/OpenTelemetryPropagatorsEventSource.cs:29]: nit:

`public static readonly OpenTelemetryPropagatorsEventSource Log = new();`

or

`public static OpenTelemetryPropagatorsEventSource Log { get; } = new();`

**Review by CodeBlanch — APPROVED** [review]
LGTM

**Comment by cijothomas** [comment]
Both the nits should be applied across entire set of EventSource classes. Would make it a follow up PR.

File: src/OpenTelemetry.Extensions.Propagators/OpenTelemetryPropagatorsEventSource.cs [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+            }
+        }
+
+        [Event(1, Message = "Failed to extract activity context in format: '{0}', context: '{1}'.", Level = EventLevel.Warning)]
+        public void FailedToExtractActivityContext(string format, string exception)
+        {
+            this.WriteEvent(1, format, exception);
+        }
+
+        [Event(1, Message = "Failed to inject activity context in format: '{0}', context: '{1}'.", Level = EventLevel.Warning)]
```
Comment (alanwest) [inline src/OpenTelemetry.Extensions.Propagators/OpenTelemetryPropagatorsEventSource.cs:46]:
```suggestion
        [Event(2, Message = "Failed to inject activity context in format: '{0}', context: '{1}'.", Level = EventLevel.Warning)]
```

**Review by alanwest — APPROVED** [review]
Tiny correction needed but 👍

**Pull request merged**